framework/encode/custom_vulkan_command_buffer_util.cpp

```diff
@@ -0,0 +1,195 @@
+/*
+** Copyright (c) 2022 LunarG, Inc.
+**
+** Permission is hereby granted, free of charge, to any person obtaining a
+** copy of this software and associated documentation files (the "Software"),
+** to deal in the Software without restriction, including without limitation
+** the rights to use, copy, modify, merge, publish, distribute, sublicense,
+** and/or sell copies of the Software, and to permit persons to whom the
+** Software is furnished to do so, subject to the following conditions:
+**
+** The above copyright notice and this permission notice shall be included in
+** all copies or substantial portions of the Software.
+**
+** THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+** IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+** FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+** AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+** LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+** FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+** DEALINGS IN THE SOFTWARE.
+*/
+
+#include "generated/generated_vulkan_command_buffer_util.h"
+
+#include "encode/vulkan_handle_wrapper_util.h"
+#include "encode/vulkan_state_info.h"
+
+#include <iostream>
+
+GFXRECON_BEGIN_NAMESPACE(gfxrecon)
+GFXRECON_BEGIN_NAMESPACE(encode)
+
+void TrackCmdPushDescriptorSetKHRHandles(CommandBufferWrapper*       wrapper,
+                                         VkPipelineLayout            layout,
+                                         uint32_t                    descriptorWriteCount,
+                                         const VkWriteDescriptorSet* pDescriptorWrites)
+{
+    assert(wrapper);
+
+    if (layout)
```

Comment:
Can this and the other tests of a handle or pointer in a boolean context compare against the null object?  Then it will be similar to the generated `TrackCmd` functions and the hand-written `Track` commands in `framework/encode/vulkan_state_tracker.cpp`.

Reply:
Done